src/ocean/task/TaskPool.d

```diff
@@ -274,3 +315,47 @@ unittest
     test!("==")(RecycleTask.recycle_count, 2,
         "RecycleTask.recycle was not called the correct number of times");
 }
+
+unittest
+{
+    // Test for waiting until all running tasks of a task pool finish executing.
+
+    static class AwaitTask : Task
+    {
+        static int value;
+
+        public void copyArguments ( )
+        {
+        }
+
+        override public void run ( )
+        {
+            value = 1;
+            .wait(1);
+            value = 2;
+        }
+    }
+
+    class MainTask : Task
+    {
+        TaskPool!(AwaitTask) my_task_pool;
+
+        public this ( )
+        {
+            this.my_task_pool = new TaskPool!(AwaitTask);
+        }
+
+        override protected void run ( )
+        {
+            this.my_task_pool.start();
```

Comment:
Good idea.

Wouldn't it be even better if I incremented `value` by one in each task? Then the final value should be equal to the number of times I called `this.my_task_pool.start()`. I'll push a fixup commit with this change, and you can tell me what you think about it.